packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-self-reg/go/integrationsteps.md

```diff
@@ -0,0 +1,292 @@
+
+if enrollResponse.Token() != nil {
+ //Since the phone factor is available in this use case, tokens won't exist until the
+ //user skips the phone factor in the next step
+}
+
+enrollResponse, err = enrollResponse.ConfirmEmail(r.Context(), r.FormValue("code"))
+
+ //Identify what is next to proceed with the register
+enrollResponse, err = enrollResponse.WhereAmI(r.Context())
+
+s.cache.Set("enrollResponse", enrollResponse, time.Minute*5)
+http.Redirect(w, r, "/enrollFactor", http.StatusFound)
+```
+
+### Step 9: Again show the list containing the available factors to enroll
+
+The next step is to show a list of available factors using the same page created in
+Step 6. Based on how you configured the Okta org for this use case, only the phone
+factor should be displayed. The `EnrollmentResponse` object's `HasStep` method you called in
+[Step 6](#step-6-build-a-list-of-available-factors-to-display-to-user) is used to toggle the visibility of the skip button and show the avaiable factors. In this step, the skip button and phone factor option should be visible.
+
+
+```go
+if enrollResponse.HasStep(idx.EnrollmentStepSkip) {
+ //This method should return true
+}
+
+if enrollResponse.HasStep(idx.EnrollmentStepPhoneVerification) {
+ //This call should return true
+}
+
+if enrollResponse.HasStep(idx.EnrollmentStepEmailVerification) {
+ //This call should return false because the email was already enrolled
+}
+```
+
+<div class="common-image-format">
+
+![A screenshot of the phone factor with skip buttin in the Golang application](/img/oie-embedded-sdk/oie-embedded-sdk-go-use-case-self-reg-phone-factor-skip.png)
+
+</div>
+
+### Step 10: Skip the phone factor
+
+Assuming the user skips the phone factor and completes the registration with only the email,
+call the `EnrollmentResponses` object's `Skip` method.
+
+```go
+func (s *Server) transitionToProfile(er *idx.EnrollmentResponse, w http.ResponseWriter, r *http.Request) {
+  session, err := sessionStore.Get(r, "direct-auth")
+
+  enrollResponse, err := er.Skip(r.Context())
+
+  ...
+```
+
+For more details about enrolling the phone factor see the sample application. For details on how
+to verify a sign-in with the phone factor, see
+[Sign in with password and phone factors](/docs/guides/oie-embedded-sdk-use-cases/go/oie-embedded-sdk-use-case-sign-in-pwd-phone/).
+
+### Step 11: Store the tokens in session and go to the signed-in home page
+
+The `EnrollmentResponse` object returned from the `Skip` method should return tokens
+indicating the register and sign-in was successful. Send the user to their
+signed-in home page.
+
+```go
+if enrollResponse.Token() != nil {
+  session.Values["access_token"] = enrollResponse.Token().AccessToken
+  session.Values["id_token"] = enrollResponse.Token().IDToken
+  err = session.Save(r, w)
+  if err != nil {
+    log.Fatalf("could not save access token: %s", err)
+  }
+}
+```
+
+### Step 12 (Optional) Retrieve the user profile information
```

Comment:
Colon required after (Optional), for consistency with other steps.

Reply:
Colon added.